finetuner/utils.py

```diff
@@ -0,0 +1,198 @@
+        the CSV are skipped. a sampling rate of 1 means that none are skipped, 0.5
+        means that half are skipped, and 0 means that all lines are skipped.
+    :param dialect: A description of the expected format of the CSV, can either be an
+        object of the :class:`csv.Dialect` class, or one of the strings returned by the
+        :meth:`csv.list_dialects()' function.
+    :param encoding: The encoding of the CSV file.
+    :param is_labeled: Whether the second column of the CSV represents a label that
+        should be assigned to the item in the first column (True), or if it is another
+        item that should be semantically close to the first (False).
+    :param convert_to_blob: Whether uris to local files should be converted to blobs
+    """
+
+    size: Optional[int] = None
+    sampling_rate: Optional[float] = None
+    dialect: Union[str, 'csv.Dialect'] = 'auto'
+    encoding: str = 'utf-8'
+    is_labeled: bool = False
+    convert_to_blob: bool = True
+
+
+def build_dataset(
+    data: Union[str, TextIO, DocumentArray],
+    model: str,
+    csv_options: Optional[CSVOptions],
+) -> Union[Generator['Document', None, None], DocumentArray]:
+
+    if isinstance(data, (TextIO)) or (isinstance(data, str) and isfile(data)):
+        if not csv_options:
+            csv_options = CSVOptions()
+
+        model_stub = get_stub(model, select_model='clip-text')
+        if csv_options:
+            data = DocumentArray(
+                load_finetune_data_from_csv(
+                    file=data,
+                    task=model_stub.task,
+                    options=csv_options,
+                )
+            )
+        else:
+            data = DocumentArray(
+                load_finetune_data_from_csv(file=data, task=model_stub.task)
+            )
+
+    return data
+
+
+def load_finetune_data_from_csv(
+    file: Union[str, TextIO],
+    task: str = 'text-to-text',
+    options: CSVOptions = CSVOptions(),
+) -> Generator['Document', None, None]:
+    """
+    Takes a CSV file and returns a generator of documents, with each document containing
+    the information from one line of the CSV.
+
+    :param file: Either a filepath to or a stream of a CSV file.
+    :param task: Specifies the modalities of the model that the returned data is to
+        be used for. This data is retrieved using the model name, and does not need
+        to be added to the csv_options argument when calling :meth:`finetuner.fit`
+    :param options: A :class:`CSVOptions` object.
+    :return: A generator of :class:`Document`s. Each document represents one element
+        in the CSV
+
+    """
+
+    if hasattr(file, 'read'):
+        file_ctx = nullcontext(file)
+    else:
+        file_ctx = open(file, 'r', encoding=options.encoding)
+
+    with file_ctx as fp:
+        # when set to auto, then sniff
+        try:
+            if isinstance(options.dialect, str) and options.dialect == 'auto':
```

Comment:
the dialect can also be of type `csv.Dialect`

Reply:
This line is specifically for the case where `dialect = 'auto'`, if it is already of type `csv.Dialect` then nothing needs to be done here